Alesia Yakutsionak/0/run.rb

```diff
@@ -0,0 +1,155 @@
+require 'rubygems'
+require 'bundler/setup'
+
+Bundler.require(:default)
+
+class ProductPrice
+  attr_accessor :min_price, :max_price, :last_price, :min_price_date, :max_price_date
+
+  def initialize(price, date)
+    @min_price = price
+    @min_price_date = date
+    @max_price = price
+    @max_price_date = date
+    @last_price = price if date == '2018/10'
+  end
+
+  def update(price, date)
+    @last_price = price if date == '2018/10'
+    if min_price > price
+      @min_price = price
+      @min_price_date = date
+    end
+    if max_price < price
+      @max_price = price
+      @max_price_date = date
+    end
+  end
+end
+
+class PriceCollector
+  MONTHS = %w(январь февраль март апрель май июнь июль август сентябрь октябрь ноябрь декабрь).freeze
+  attr_accessor :products
+
+  def initialize
+    @products = {}
+  end
+
+  def start
+    collect_data
+    found = search_product(ask_product)
+    found.each do |product|
+      show_found_product(product)
+      show_same_price_products(found, product)
+      puts
+    end
+  end
+
+  private
+
+  def collect_data
+    print 'Please wait'
+    Dir['data/*'].each do |f|
+      if f.end_with?('.xls')
+        collect_from_xls(f)
+      elsif f.end_with?('.xlsx')
+        collect_from_xlsx(f)
+      end
+      print '.'
+    end
+    puts
+  end
+
+  def collect_from_xls(file)
+    xls = Roo::Excel.new(file)
+    sheet = xls.worksheets.first
+    date = parse_date(sheet.rows[2].compact.first)
+    puts "Can't parse date in #{file}" unless date
+
+    sheet.rows.each_with_index do |row, i|
+      next if i < 8 || row[0].nil?
+      name = clean_name(row[0])
+      price = row[6]
+      add_product(name, price, date)
+    end
+  end
+
+  def collect_from_xlsx(file)
+    xlsx = Roo::Excelx.new(file)
+    date = parse_date(xlsx.sheet(0).row(3).compact.first)
+    puts "Can't parse date in #{file}" unless date
+
+    xlsx.sheet(0).each_row_streaming(offset: 7) do |row|
+    name = clean_name(row[0].cell_value)
+      next if name.nil?
+      price = row[6].cell_value
+      add_product(name, price, date)
+    end
+  end
+
+  def add_product(name, price, date)
+    price = price.to_f
+    price /= 10_000 if date < '2017/01'

```

Comment:
make 10000, 2017/01 constants
also you can add helper method
```
def convert_price(price, date)
...
price = convert_price(price, date)
```
that will handle both denomination and rounding